2_process/src/do_lakegroup_tasks.R

```diff
@@ -67,14 +58,38 @@ create_group_tasks <- function(task_ids, log_folder, irradiance_zips, clarity_zi
     } 
   )
   
-  # ---- combine into a task plan ---- #
+  ##-- Create the task plan --##
   
-  scipiper::create_task_plan(
+  task_plan <- create_task_plan(
     task_names=tasks$task_id,
     task_steps=list(
       get_lake_ids,
       combine_model_zipped_files),
     add_complete=FALSE,
-    final_steps=c('combine_model_zipped_files'),
-    ind_dir='2_process/log')
+    final_steps=c('combine_model_zipped_files'))
+  
+  ##-- Create the task remakefile --##
+  
+  task_makefile <- sprintf('2_%s_grp_tasks.yml', model_prefix)
+  create_task_makefile( 
+    task_plan=task_plan,
+    makefile=task_makefile,
+    include='remake.yml',
+    sources=c(...),
+    packages=c("purrr", "dplyr", "readr", "feather"),
+    final_targets = final_target,
+    finalize_funs = "indicate_file_dataframes",
+    as_promises = TRUE,
+    tickquote_combinee_objects = TRUE
+  )
+  
+  ##-- Build the tasks --##
+  
+  loop_tasks(task_plan = task_plan,
+             task_makefile = task_makefile,
+             num_tries = 3)
+  
+  # Now return the file name of the final target
+  return(remake::fetch(sprintf("%s_promise", basename(final_target)), remake_file=task_makefile))
```

Comment:
Since fetch doesn't do a build but instead grabs the file location or object from remake's database, isn't this going to fetch the `promise` _object_ that is kind of a dummy target and return that? Instead, can/should you `return(final_target)` in these cases?

Reply:
I think I was following pipelines-3 in this pattern, but I realize now that was a special use case. We wanted to return multiple things and we wanted the objects returned. There weren't any files.

Yes, seems like just returning the final target is what I should be doing!

Reply:
Should you be returning anything at all? This function is used to produce a file, right? So remake is checking for the creation of the file, not the contents of a (constant) filename string returned by the function.

Reply:
@aappling-usgs you are absolutely right. I don't think I need to return anything. Feels kind of weird to write a fxn without `return()`!